src/block/Wall.php

```diff
@@ -70,42 +71,129 @@ public function setPost(bool $post) : self{
 	}
 
 	public function onNearbyBlockChange() : void{
-		if($this->recalculateConnections()){
+		$connectionsUpdated = $this->recalculateConnections();
+		$postUpdated = $this->recalculatePost();
+		if($connectionsUpdated || $postUpdated){
 			$this->position->getWorld()->setBlock($this->position, $this);
 		}
 	}
 
 	protected function recalculateConnections() : bool{
-		$changed = 0;
-
-		foreach(Facing::HORIZONTAL as $facing){
-			$block = $this->getSide($facing);
-			if($block instanceof static || $block instanceof FenceGate || $block instanceof Thin || $block->getSupportType(Facing::opposite($facing)) === SupportType::FULL){
-				$aboveBlock = $this->getSide($facing)->getSide(Facing::UP);
-				$connectionType = ($aboveBlock->getTypeId() === BlockTypeIds::AIR)
-					? WallConnectionType::SHORT
-					: WallConnectionType::TALL;
-
-				if(!isset($this->connections[$facing]) || $this->connections[$facing] !== $connectionType){
-					$this->connections[$facing] = $connectionType;
-					$changed++;
+		$updated = false;
+		$above = $this->getSide(Facing::UP);
+		$abovePos = $above->getPosition();
+
+		foreach(Facing::HORIZONTAL as $face) {
+			$side = $this->getSide($face);
+			$sidePos = $side->getPosition();
+
+			// TODO: Improve stair check by checking corners
+
+			$connected = match (get_class($side)) {
+				Cake::class => false,
+				Campfire::class => Facing::opposite($face) === Facing::DOWN,
+				Fence::class => Facing::opposite($face) === Facing::DOWN || Facing::opposite($face) === Facing::UP,
+				Hopper::class => Facing::opposite($face) === Facing::UP,
+				Slab::class => $side->getSlabType() === SlabType::DOUBLE || ($side->getSlabType() === SlabType::TOP && Facing::opposite($face) === Facing::UP) || Facing::opposite($face) === Facing::DOWN,
+				Stair::class => (!$side->isUpsideDown() && Facing::opposite($face) === Facing::DOWN) ||
```

Comment:
No way, there has to be a better way than hardcoding logic for every block...

Reply:
The supportType can probably be used here (juste the idea, i don't have checked)